ipns/name.go

```diff
@@ -25,7 +25,7 @@ const (
 // [Multihash]: https://multiformats.io/multihash/
 // [IPNS Name]: https://specs.ipfs.tech/ipns/ipns-record/#ipns-name
 type Name struct {
-	src []byte
+	multihash string // binary Multihash without multibase envelope
```

Comment:
:information_source:  I've pushed small change  (8c1d32a0fb01e2832d76a0fe9bf292987cc1bf73) that renamed `src` to meaningful `multihash`  and added  the above comment, so at least a drive-by reader does not make wrong assumptions.

fwiw I looked across ecosystem, our core types are mostly `string`:

- go-multihash `type Multihash []byte`
- go-cid `type Cid struct{ str string }`
- go-libp2p/peer  `type ID string`

I guess `string` here is ok, especially now that we've added comment, but 

@hacdias  did we switch from `[]byte` to `string` here   to allow them in map keys, or was there a deeper reason?